### ISSUE TITLE
unable to install quippy-ase through either pip install or manual installation

### ISSUE THREAD

**Issue by GUANGZChen** [libAtoms/QUIP#572]
Hi there,

I just fit a GAP using a current version of gap_fit. When I tried to load the potential via quippy-ase installed by pip (`pip install quippy-ase`), it didn't work with the error message: `'Database was created with a later version of the code. Version of code used to generate the database is 1675289491. Version of current code is 1673447045. Please update your code.'`

Then I tried the alternative way to install quippy manually. It came out with the error:`'Encountered error while generating package metadata python setup.py egg_info did not run successfully.'` and `'the version specified is an invalid version, this may not work as expected with newer versions of setuptools, pip, and pypi. please see pep 440 for more details.'`

A more detailed error message is given here:
```
Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [28 lines of output]
      version: 0.9.12-14-gb55e89d55-dirty
      running egg_info
      /home/zhao/anaconda3/envs/cgz38/lib/python3.8/site-packages/setuptools/dist.py:543: UserWarning: The version specified ('0.9.12-14-gb55e89d55-dirty') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
        warnings.warn(
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/home/zhao/code/QUIP_2023.3.7/build/linux_x86_64_gfortran/setup.py", line 52, in <module>
          setup(
        File "/home/zhao/anaconda3/envs/cgz38/lib/python3.8/site-packages/setuptools/__init__.py", line 87, in setup
          return distutils.core.setup(**attrs)
        File "/home/zhao/anaconda3/envs/cgz38/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/home/zhao/anaconda3/envs/cgz38/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/home/zhao/anaconda3/envs/cgz38/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/home/zhao/anaconda3/envs/cgz38/lib/python3.8/site-packages/setuptools/dist.py", line 1208, in run_command
          super().run_command(command)
        File "/home/zhao/anaconda3/envs/cgz38/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.ensure_finalized()
        File "/home/zhao/anaconda3/envs/cgz38/lib/python3.8/site-packages/setuptools/_distutils/cmd.py", line 111, in ensure_finalized
          self.finalize_options()
        File "/home/zhao/anaconda3/envs/cgz38/lib/python3.8/site-packages/setuptools/command/egg_info.py", line 219, in finalize_options
          parsed_version = parse_version(self.egg_version)
        File "/home/zhao/anaconda3/envs/cgz38/lib/python3.8/site-packages/pkg_resources/_vendor/packaging/version.py", line 266, in __init__
          raise InvalidVersion(f"Invalid version: '{version}'")
      pkg_resources.extern.packaging.version.InvalidVersion: Invalid version: '0.9.12-14-gb55e89d55-dirty'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
```

It is much appreciated if someone could help.

**Comment by jameskermode** [comment]
You can probably just revert the version number in your fitted XML file to match the version of the released quippy-ase package (1673447045). This will be fine provided there haven't been any backwards-incompatible changes to gap_fit since the release in January, which from memory should be OK. If this doesn't work let me know and we can do a new release. 

@albapa @Sideboard we could try to think of a better solution to this issue, since it comes up more now that we do (somewhat infrequent) binary releases of quippy-ase. We could just do a new release after merging every PR, this wouldn't be a big job, and could even be automated.

**Comment by GUANGZChen** [comment]
Thanks! It works after changing the version number.
